### ISSUE TITLE
Inheritance generated constructors can clash with other constructors

### ISSUE THREAD

**Issue by beikov** [Blazebit/blaze-persistence#480]
The test case was submitted via: https://github.com/Blazebit/blaze-persistence/pull/464

The problem is, that the "default" constructor generated for BView clashes with the constructor generated for inheritance. This is because the attribute names are in the same order for both cases, the subtype configuration and the "default non-subtype case".
The underlying problem is that type clashes could also happen becuase of reorderings.
Imagine AView having a String. BView extends AView and also has a String. The default constructor might be like (AView.String, BView.String) whereas the subtype configuration constructor might be (BView.String, AView.String). Simply omitting the type clashing constructor wouldn't work as the sematics of the constructors are different.

I'm not yet sure how to approach this, but I have a feeling that it might be better to have just a single constructor that has a total order of attributes and instead create static factory methods that map to that single constructor. I just don't see how we could solve this by using just a constructor.

**Comment by JWGmeligMeyling** [comment]
Why have multiple constructors at all? Can’t we annotate the constructor parameters with some mapping meta data such as the mapping field and its originating view class, so that we can resolve the parameters from there?

**Comment by beikov** [comment]
I did somewhat similar to what you suggested but without the annotations.
The final design I came up with that I'm currently trying to implement is the following...

An entity view class hierarchy might have multiple usages. When a type of the hierarchy is used statically(attribute type), it becomes an _inheritanceBase_. The effect is, that every subtype of that _inheritanceBase_ class gets a specific class generated that is bound to that _inheritanceBase_.
The next concept that is interesting is about _subtype configurations_. In JPA you only have a single subtype configuration i.e. every entity subtype of an inheritance enabled entity is part of that subtype configuration. In Blaze-Persistence Entity Views, we allow to have multiple subtype configurations by specifying the interesting subtypes at the use site(attribute declaration).
At runtime, every subtype configuration needs an appropriate _factory_ method that is used to construct subtype instances uniformly. Previously, we tried to generate constructors for every configuration which doesn't work out. The new idea is, to construct an overall configuration that contains all subtypes for which only a single constructor is created. Every other subtype configuration has a mapping to that overall configuration. For each subtype configuration, we now generate a _static create_ method that maps the configuration specific attributes to the overall constructor. That way, we avoid clashing constructors when having multiple configurations. The _static create_ methods are suffixed with their subtype configuration index.
